tests/plugins/lti_consumer/test_forms.py

```diff
@@ -164,3 +165,33 @@ def test_forms_lti_consumer_url_not_regex_included(self):
                 ]
             },
         )
+
+    @override_settings(RICHIE_LTI_PROVIDERS=get_lti_settings())
+    def test_forms_lti_consumer_reset_credentials(self):
+        """
+        The "oauth_consumer_key" and "shared_secret" fields should be reset when a value
+        is set for the "lti_provider_id" field.
+        """
+        lti_consumer = LTIConsumerFactory(
+            lti_provider_id=None,
+            oauth_consumer_key="OauthConsumerKey",
+            shared_secret="SharedSecret",
+        )
+        self.assertIsNotNone(lti_consumer.oauth_consumer_key)
+        self.assertIsNotNone(lti_consumer.shared_secret)
+
```

Comment:
In fact I'm checking that the `shared_secret` field is not None because we are tickering with field values in the `save` method and I want to make sure it only goes to None at the moment I want it to go to None. I'm making sure it was not already None from the start ^^
I'm not sure what improvement you are suggestion. Can you make a code suggestion?

Reply:
I started a new conversation below to be able to give a suggestion :-).